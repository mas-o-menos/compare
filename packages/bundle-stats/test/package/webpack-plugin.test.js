const webpack = require('webpack');
const MemoryFS = require('memory-fs');
const config = require('./app/webpack.config');

jest.setTimeout(10 * 1000);

test('webpack plugin', (done) => {
  expect.assertions(3);

  const compiler = webpack(config);
  compiler.outputFileSystem = new MemoryFS()

  compiler.run((error, stats) => {
    expect(error).toEqual(null);
    expect(stats.hasErrors()).toBe(false);
    expect(stats.toJson({ source: false, assets: true }).assets).toMatchSnapshot();
    done();
  });
});
