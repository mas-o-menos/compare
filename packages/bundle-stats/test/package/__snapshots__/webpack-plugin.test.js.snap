// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack plugin package 1`] = `
Array [
  Object {
    "chunkNames": Array [],
    "chunks": Array [],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "bundle-stats.html",
    "size": 0,
  },
  Object {
    "chunkNames": Array [
      "main",
    ],
    "chunks": Array [
      0,
    ],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "main.js",
    "size": 956,
  },
]
`;
